### PR TITLE
[panorama] PanoramaCompositing: new out-of-core algorithm

### PR DESCRIPTION
**Pull request by servantf**
These updates enable the creation of very large panoramas.

--> optimized algorithms to work on tiles
--> creation of disk cache to remove the RAM barrier

### CODE REVIEW

File: src/aliceVision/image/cache.hpp [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+  void addFreeBlock(size_t blockId, size_t blockCount);
+  size_t getFreeBlockId(size_t blockCount);
+
+  
+
+protected:
+  size_t _blockSize;
+  size_t _incoreBlockUsageCount;
+  size_t _incoreBlockUsageMax;
```
Comment (fabiencastan) [inline src/aliceVision/image/cache.hpp:238]: Add some doxygen doc to explain the CacheManager params.
For instance, I'm not sure what is _incoreBlockUsageMax.

File: src/software/pipeline/main_panoramaCompositing.cpp [hunk 2]
```diff
@@ -41,2459 +44,602 @@ namespace po = boost::program_options;
...
+    
+    bool succeeded = true;
 
-  if (AminusB.size() != B.size()) {
-    return false;
-  }
+    omp_set_num_threads(std::min(omp_get_thread_limit(), std::max(0, maxThreads)));
 
-  for (int i = 0; i < height; i++) {
+    //#pragma omp parallel for
```
Comment (fabiencastan) [inline src/software/pipeline/main_panoramaCompositing.cpp:624]: To remove or to uncomment?

File: src/software/pipeline/main_panoramaCompositing.cpp [hunk 3]
```diff
@@ -41,2459 +44,602 @@ namespace po = boost::program_options;
...
+    }
+    
+    const std::vector<IndexT> & chunk = chunks[rangeIteration];
+    
+    bool succeeded = true;
 
-  if (AminusB.size() != B.size()) {
-    return false;
-  }
+    omp_set_num_threads(std::min(omp_get_thread_limit(), std::max(0, maxThreads)));
```
Comment (fabiencastan) [inline src/software/pipeline/main_panoramaCompositing.cpp:622]:
```suggestion
    if(maxThreads > 0)
        omp_set_num_threads(std::min(omp_get_thread_limit(), maxThreads));
```

Comment (servantf) [reply src/software/pipeline/main_panoramaCompositing.cpp:622]: It means that 0 would equals to auto ?

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**